### ISSUE TITLE
Compilation not completed at task completion

### ISSUE THREAD

**Issue by staff0rd** [ivogabe/gulp-typescript#263]
Am I incorrectly expecting typescript compilation to be completed on task completion?  For example, I have the following in [gulpfile.js](https://github.com/staff0rd/gulp-typescript-test/blob/master/gulpfile.js);

~~~javascript
gulp.task('watch', ['compile'], function () {
  gulp.watch('./src/**/*.ts', ['compile']);
  gulp.watch('./src/compiled.js', ['reload']);
});

gulp.task('compile', function(){
   gulp.src('./src/**/*.ts')
     .pipe(typescript({out: 'compiled.js'}))
     .pipe(gulp.dest('./src'))
});
~~~

Which in my (admittedly beginner) understanding means `compile` will be completed prior to `watch`, therefore on gulp startup if I call `watch`, `reload` is not immediately triggered.

However, I am finding that `reload` is being triggered.  Should `typescript()` not be completed by the time the second `watch` is applied to trigger `reload`, and as such, `reload` not be called?

~~~
c:\git\gulp-typescript-test>gulp
[12:17:12] Using gulpfile c:\git\gulp-typescript-test\gulpfile.js
[12:17:12] Starting 'connect'...
[12:17:12] Finished 'connect' after 31 ms
[12:17:12] Starting 'compile'...
[12:17:12] Finished 'compile' after 9.63 ms
[12:17:12] Starting 'watch'...
[12:17:12] Finished 'watch' after 14 ms
[12:17:12] Starting 'default'...
[12:17:12] Finished 'default' after 4.46 μs
[12:17:12] Server started http://localhost:9000
[12:17:12] LiveReload started on port 35729
[12:17:13] Starting 'reload'...
[12:17:13] Finished 'reload' after 14 ms
~~~

[Repro here](https://github.com/staff0rd/gulp-typescript-test).

**Comment by staff0rd** [comment]
This seems additionally highlighted by the fact that compilation errors are displayed in the terminal after `Finished 'compile'` is printed.

**Comment by ivogabe** [comment]
You should return the stream in a task, like this:
```javascript
gulp.task('compile', function(){
   return gulp.src('./src/**/*.ts')
     .pipe(typescript({out: 'compiled.js'}))
     .pipe(gulp.dest('./src'))
});
```

**Comment by staff0rd** [comment]
This of course worked as expected, thanks so much.
